### PR TITLE
Add missing window flags for styling system bars

### PR DESCRIPTION
**Pull request by camsim99**
Addresses the inability to set the icon brightness on system bars on API 30, part of issue [#10027](https://github.com/flutter/flutter/issues/100027).

Using the [reproduction](https://github.com/flutter/flutter/issues/100027#issue-1167643363) included in the issue noted above, I was able to produce the following on a Pixel 2 emulator running API 30 with this change (see before [here](https://github.com/flutter/flutter/issues/100027#issue-1167643363)):

<p align="center">

</p>

I confirmed these results on Pixel devices running API 28 (with [this revert](https://github.com/flutter/flutter/pull/100263) included), 29, and 31.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by Piinks — COMMENTED** [review]
Can you add a test that verifies these flag settings independent of the divider color being set?

**Comment by zanderso** [comment]
From PR review triage: This looks ready for another review pass @blasten @Piinks

File: shell/platform/android/io/flutter/plugin/platform/PlatformPlugin.java [hunk 1]
```diff
@@ -370,6 +370,16 @@ private void setSystemChromeSystemUIOverlayStyle(
...
         new WindowInsetsControllerCompat(window, view);
 
+    // Flag set to specify that this window is responsible for drawing the background for the system
+    // bars.
+    window.addFlags(WindowManager.LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS);
+
+    // Flag set to dismiss any requests for translucent system bars to be provided in lieu of what
+    // is specified by systemChromeStyle.
+    window.clearFlags(
+        WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS
```
Comment (blasten) [inline shell/platform/android/io/flutter/plugin/platform/PlatformPlugin.java:380]: this is deprecated in API 30

File: shell/platform/android/io/flutter/plugin/platform/PlatformPlugin.java [hunk 2]
```diff
@@ -370,6 +370,16 @@ private void setSystemChromeSystemUIOverlayStyle(
...
+            | WindowManager.LayoutParams.FLAG_TRANSLUCENT_NAVIGATION);
+
```
Comment (blasten) [inline shell/platform/android/io/flutter/plugin/platform/PlatformPlugin.java:382]: These flags are deprecated in API 30

**Review by blasten — COMMENTED** [review]
What does the status bar look like on API 27?

File: shell/platform/android/io/flutter/plugin/platform/PlatformPlugin.java (hunk 2 above)
Comment (camsim99) [reply shell/platform/android/io/flutter/plugin/platform/PlatformPlugin.java]: They are needed to fix the problem for API < 30, so I added a comment noting the deprecation and suppressed the warning

**Comment by camsim99** [comment]
> What does the status bar look like on API 27?

It looks the same as the pictures above! I also tested this on API 21 to see what happens when none of the operations can be performed (API 23 is needed at least), and it looks the same with and without setting/clearing the flags. I can also move the setting/clearing of flags after the SDK versions are checked for each operation (like [here](https://github.com/flutter/engine/pull/32167/files#diff-8dfd4b72efbe5f5b494de945a284feba1d7e63a71e56a567235e6748717801caL436)) to rule out side effects, but I was just trying to avoid redundant code.

**Review by blasten — APPROVED** [review]
LGTM

**Pull request merged**